core.spec.md

```diff
@@ -249,6 +250,36 @@ directive @eg(data: eg__Data) on FIELD_DEFINITION
 directive @core(feature: String!, as: String) repeatable on SCHEMA
 ```
 
+###! for: [core__Purpose!]
+
+An optional list of [purposes](#core__Purpose) for this feature. This provides a hint to processors and data cores as to whether they can safely ignore a given feature.
+
+By default, core features SHOULD fail open. This means that an unknown feature SHOULD NOT prevent a schema from being served or processed. Instead, consumers SHOULD ignore unknown features and serve or process the rest of the schema normally.
+
+This behavior is different for features with a specified purpose. Currently, the only specified purpose is [`security`](#core__Purpose/security). Thus, the acceptable values for this argument are `for: [security]` or {null}. Data cores MUST NOT attempt to serve a schema unless they support **all** features which are referenced `for` [`security`](#core__Purpose/security).
+
+Note technically it is valid to repeat values within an enum list, so `[security, security]`, `[security, security, security]` and so on are valid argument values. This is not an error, and SHOULD be treated as equivalent to just `[security]`. However, core schema authors SHOULD avoid doing this, as it does nothing but add noise to the document.
+
+# Enums
+
+##! core__Purpose
+
+```graphql definition
+enum core__Purpose {
+  security
```

Comment:
I agree with @glasser that it could make sense to categorize these by processor behavior rather than purpose. My first response was also that something like `serviceMustSupport` would be clearer, but we may want to think about the kinds of features and expected behavior first.